app-editors/atom/atom-1.27.2.ebuild

```diff
@@ -0,0 +1,563 @@
+
+	# Remove non-Linux vendored ctags binaries
+	rm "${BUILD_DIR}/app/${ctags_d}/ctags-darwin" \
+	   "${BUILD_DIR}/app/${ctags_d}/ctags-win32.exe" || die
+
+	# Re-pack app.asar
+	# Keep unpack rules in sync with build/tasks/generate-asar-task.coffee
+	cd "${BUILD_DIR}" || die
+	x="--unpack={*.node,ctags-config,ctags-linux,**/spec/fixtures/**,**/node_modules/spellchecker/**,**/resources/atom.png}"
+	xd="--unpack-dir=apm"
+	easar pack "${x}" "${xd}" "app" "app.asar"
+
+	rm -r "${BUILD_DIR}/app.asar.unpacked/apm" || die
+
+	# Replace vendored ctags with a symlink to system ctags
+	rm "${BUILD_DIR}/app.asar.unpacked/${ctags_d}/ctags-linux" || die
+	ln -s "${EROOT%/}/usr/bin/ctags" \
+		"${BUILD_DIR}/app.asar.unpacked/${ctags_d}/ctags-linux" || die
+}
+
+src_test() {
+	local electron="$(get_electron_dir)/electron"
+	local app="${BUILD_DIR}/app.asar"
+
+	virtx "${electron}" --app="${app}" --test "${app}/spec"
+}
+
+src_install() {
+	local install_dir="$(get_install_dir)"
+	local suffix="$(get_install_suffix)"
+
+	insinto "${install_dir}"
+
+	doins "${BUILD_DIR}/app.asar"
+	doins -r "${BUILD_DIR}/app.asar.unpacked"
+
+	insinto "${install_dir}/app"
+	doins -r "${BUILD_DIR}/app/apm"
+
+	insinto "/usr/share/applications/"
+	newins "${BIN_S}/usr/share/applications/$(get_atom_appname).desktop" \
+		"atom${suffix}.desktop"
+
+	insinto "/usr/share/icons/"
+	doins -r "${BIN_S}/usr/share/icons/hicolor"
+
+	exeinto "${install_dir}"
+	newexe "${BUILD_DIR}/app/atom.sh" atom
+	insinto "/usr/share/licenses/${PN}${suffix}"
+	doins "${BIN_S}/$(get_atom_rpmdir)/resources/LICENSE.md"
+	dosym "${install_dir}/atom" "/usr/bin/atom${suffix}"
+	dosym "${install_dir}/app/apm/bin/apm" "/usr/bin/apm${suffix}"
+
+	_fix_executables "${install_dir}/app/apm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/.bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin/node-gyp-bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/node-gyp/bin"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
+
+# Helpers
+# -------
+
+# Return the installation suffix appropriate for the slot.
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
+}
+
+# Return the upstream app name appropriate for $PV.
+get_atom_appname() {
+	if [[ "${PV}" == *beta* ]]; then
+		echo -n "atom-beta"
+	else
+		echo -n "atom"
+	fi
+}
+
+# Return the app installation path inside the upstream archive.
+get_atom_rpmdir() {
+	echo -n "usr/share/$(get_atom_appname)"
+}
+
+# Return the installation target directory.
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+# Return the Electron installation directory.
+get_electron_dir() {
+	echo -n "${EROOT%/}/usr/$(get_libdir)/electron-${ELECTRON_SLOT}"
```

Comment:
EROOT is not valid in `src_*`. It's only meaningful in `pkg_{pre,post}{inst,rm}`

Reply:
What is the correct way to get the absolute path to the electron binary installed by `dev-util/electron`?

Reply:
Putting it in `BDEPEND` and using `BROOT` (EAPI 7).

Reply:
OK, so when I'm patching the source in `src_prepare` to embed the correct paths, I should continue to use `${EROOT}`, and when I'm running electron to during build, I should use `${BROOT}`, essentially needing to have both `get_electron_broot_dir` and `get_electron_eroot_dir`.  Is that correct?

Reply:
No, 'correct paths' don't include any `*ROOT` at all. `*ROOT`s are temporarily things applying to the particular emerge execution. You're looking for `EPREFIX`.